sdk/core/azure-core-amqp/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_types.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef ANQP_TYPES_H
+#define ANQP_TYPES_H
+
+#include <stddef.h>
+#include "azure_macro_utils/macro_utils.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#define AMQP_TYPE_VALUES \
```

Comment:
How visible would this be to the SDK users? Are we going to leak any of the stuff that's defined in these headers (not this specific macro, but anything)?

If our headers do only forward-declare AMQP types at best, and it is only the SDK cpp files code that see this, then everything is easy and I am very happy.

Reply:
I.e. if you would produce an ApiView from this, without additional tweaks to ignore certain stuff, what would be visible?

Reply:
uAMQP is a vendored version of azure-uamqp-c.

It is NOT an azure SDK. 

It is literally a replacement for the azure-uamqp-c vcpkg package.

There is a fair amount of work in the azure-core-amqp to ensure that there is no leakage.